### DISCUSSION TITLE
Reproduce and avoid RESTEASY004655: Unable to invoke request: java.net.SocketException: Connection reset

### DISCUSSION THREAD

**Discussion by fernandocorrea** [resteasy/resteasy#4328] [Q&A] [answered]
Hi,

I'm getting the error "RESTEASY004655: Unable to invoke request: java.net.SocketException: Connection reset" in a low percent comparing with total number of request per day...however I still cannot reproduce this error...

I've searched around and gt to reprocude it using sockets (following this link https://www.mastertheboss.com/java/how-to-solve-the-error-java-net-socketexception-connection-reset/)
but I can't reproduce it with resteasy-client (version 4.7.1)..

Can someone help me, please?

**Comment by jamezp** [comment]
Where are you seeing these? On your server or on a client?

**Reply by fernandocorrea** [reply]
hi..
In server side

**Comment by fernandocorrea** [comment]
I managed to reproduce the error with this classes:

```
import java.io.IOException;
import java.net.ServerSocket;
import java.net.Socket;
public class Server {
    public static void main(String[] args) {
        final int PORT = 8080;

        try (ServerSocket serverSocket = new ServerSocket(PORT)) {
            System.out.println("Server listening on port " + PORT);

            while (true) {
                Socket clientSocket = serverSocket.accept();
                System.out.println("New client from " + clientSocket.getInetAddress() + ":" + clientSocket.getPort());
                Thread.sleep(1000);
                clientSocket.setSoLinger(true, 0);
                clientSocket.close();
            }
        } catch (IOException | InterruptedException e) {
            e.printStackTrace();
        }

    }
}
```
```
import org.jboss.resteasy.client.jaxrs.ResteasyClient;
import org.jboss.resteasy.client.jaxrs.ResteasyWebTarget;
import org.jboss.resteasy.client.jaxrs.internal.ResteasyClientBuilderImpl;

public class RestEasyClient {
  public static void main(String[] args) {
    ResteasyClient client = new ResteasyClientBuilderImpl().build();
    ResteasyWebTarget target = client.target("http://localhost:8080");
    target.request().get();
  }
}
```


I guess there's no way to avoid this error to happen...
after RestEasy's "automatic retries" 
if the error remains I guess there's nothing to do but, to investigate  the close cause in network side(correct me if I'm wrong)...

**Reply by jamezp** [reply, marked as answer]
Oops, my bad. I just realized I never pressed reply :)

Yes, there isn't much RESTEasy can do if a client closes the connection during a request.

**Reply by fernandocorrea** [reply]
Thanks @jamezp
